internal/service/s3/bucket_website_configuration.go

```diff
@@ -0,0 +1,613 @@
+package s3
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/verify"
+)
+
+func ResourceBucketWebsiteConfiguration() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceBucketWebsiteConfigurationCreate,
+		ReadContext:   resourceBucketWebsiteConfigurationRead,
+		UpdateContext: resourceBucketWebsiteConfigurationUpdate,
+		DeleteContext: resourceBucketWebsiteConfigurationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"bucket": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 63),
+			},
+			"error_document": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"key": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+			"expected_bucket_owner": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: verify.ValidAccountID,
+			},
+			"index_document": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"suffix": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+			"redirect_all_requests_to": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				ConflictsWith: []string{
+					"error_document",
+					"index_document",
+					"routing_rule",
+				},
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"host_name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"protocol": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringInSlice(s3.Protocol_Values(), false),
+						},
+					},
+				},
+			},
+			"routing_rule": {
+				Type:     schema.TypeSet,
```

Comment:
Is there any ordering to the redirect rules? E.g. if I create rules like

```
`docs/`        => destinationA
`docs/images/` => destinationB
```

does the order in the generated JSON matter?

Reply:
Ooo that's a good question, and i want to say it _does_ matter so this should really be a List..also the `aws_s3_bucket` resource's `routing_rules` argument doesn't have any 
```				
DiffSuppressFunc: verify.SuppressEquivalentJSONDiffs,
```
which makes me think the List is more appropriate

Reply:
updated to TypeList in [a90dcdc](https://github.com/hashicorp/terraform-provider-aws/pull/22648/commits/a90dcdcde0b0754d69550612a44e32d7791ec7bc) 👍